### PR TITLE
SUS-2973: MultiLookup cleanup and refactor

### PR DESCRIPTION
**Pull request by mszabo-wikia**
https://wikia-inc.atlassian.net/browse/SUS-2973

### CODE REVIEW

File: extensions/wikia/SpecialMultipleLookup/MultiLookupPager.php [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	 * string not &#160; for empty cells. Do not include the <td> and </td>.
+	 *
+	 * The current result row is available as $this->mCurrentRow, in case you
+	 * need more context.
+	 *
+	 * @param $name String: the database field name
+	 * @param $value String: the value retrieved from the database
+	 * @return string
+	 */
+	function formatValue( $name, $value ) {
```
Comment (WikiaTech) [inline extensions/wikia/SpecialMultipleLookup/MultiLookupPager.php:123]: MAJOR Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/SpecialMultipleLookup/MultiLookupPager.php [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	/**
+	 * Get recent contributors that used this IP address on a wiki
+	 * @param string $dbName
+	 * @return string[] array of user names and IP addresses
+	 */
+	private function getWikiUsers( $dbName ) {
+		global $wgMemc;
+
+		$cacheKey = wfSharedMemcKey( 'multilookup', $this->target, $dbName );
+		$cachedValue = $wgMemc->get( $cacheKey );
```
Comment (macbre) [inline extensions/wikia/SpecialMultipleLookup/MultiLookupPager.php]: Let's use `WikiaDataAccess` helper class.

File: extensions/wikia/SpecialMultipleLookup/MultiLookupPager.php [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+			__METHOD__,
+			[ 'DISTINCT' ]
+		);
+
+		$users = [];
+		foreach ( $res as $user ) {
+			// SUS-812: use username lookup - user ID for registered users, rc_user_text for anons
+			$userName = User::getUsername( $user->user_id, $user->user_name );
+
+			$users[] = $userName;
```
Comment (macbre) [inline extensions/wikia/SpecialMultipleLookup/MultiLookupPager.php]: Do we want this array to have unique values?

File: extensions/wikia/SpecialMultipleLookup/SpecialMultipleLookup.php [hunk 4]
```diff
@@ -11,18 +11,27 @@
...
 	'description' => 'Provides user lookup on multiple wikis',
 	'descriptionmsg' => 'specialmultiplelookup-desc',
-	'author' => array( 'Bartek Lapinski', 'Piotr Molski' ),
-	'url' => 'https://github.com/Wikia/app/tree/dev/extensions/wikia/SpecialMultipleLookup'
-);
-define( "MULTILOOKUP_NO_CACHE", false );
-define( "ML_TEST", 0 );
-require_once( dirname( __FILE__ ) . '/SpecialMultipleLookup_helper.php' );
-require_once( dirname( __FILE__ ) . '/SpecialMultipleLookup_ajax.php' );
-require_once( dirname( __FILE__ ) . '/SpecialMultipleLookup_hooks.php' );
```
Comment (macbre) [inline extensions/wikia/SpecialMultipleLookup/SpecialMultipleLookup.php]: :+1:

**Review by macbre — APPROVED** [review]
Left few minor comments / questions

File: extensions/wikia/SpecialMultipleLookup/MultiLookupPager.php (hunk 3 above)
Comment (mszabo-wikia) [reply extensions/wikia/SpecialMultipleLookup/MultiLookupPager.php]: `SELECT ... DISTINCT` should handle that part 🙂

Comment (macbre) [reply extensions/wikia/SpecialMultipleLookup/MultiLookupPager.php]: Yes, wanted to suggest the same :)

File: extensions/wikia/SpecialMultipleLookup/MultiLookupPager.php [hunk 5]
```diff
@@ -63,40 +63,32 @@ function formatRow( $row ) {
...
+			$users = [];
+			foreach ( $res as $user ) {
+				// SUS-812: use username lookup - user ID for registered users, rc_user_text for anons
+				$userName = User::getUsername( $user->user_id, $user->user_name );
+
+				$users[] = $userName;
+			}
+
+			return $users;
+		} );
```
Comment (macbre) [inline extensions/wikia/SpecialMultipleLookup/MultiLookupPager.php:93]: :+1:

**Review by macbre — APPROVED** [review]

**Review by macbre — APPROVED** [review]

**Comment by v3rth** [comment]
+381 −1,600 <3

**Pull request merged**